hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/HoodieStreamerUtils.java

```diff
@@ -88,9 +89,10 @@ public class HoodieStreamerUtils {
    */
   public static Option<JavaRDD<HoodieRecord>> createHoodieRecords(HoodieStreamer.Config cfg, TypedProperties props, Option<JavaRDD<GenericRecord>> avroRDDOptional,
                                                                   SchemaProvider schemaProvider, HoodieRecord.HoodieRecordType recordType, boolean autoGenerateRecordKeys,
-                                                                  String instantTime, Option<BaseErrorTableWriter> errorTableWriter) {
+                                                                  String instantTime, Option<BaseErrorTableWriter> errorTableWriter, HoodieTableConfig tableConfig) {
     boolean shouldCombine = cfg.filterDupes || cfg.operation.equals(WriteOperationType.UPSERT);
-    boolean shouldUseOrderingField = shouldCombine && !StringUtils.isNullOrEmpty(cfg.sourceOrderingFields);
+    String orderingFieldsStr = tableConfig.getOrderingFieldsStr().orElse(cfg.sourceOrderingFields);
```

Comment:
we don't want to make this default and siltently move on. 

lets re-use table config only for mysqldebezium payload and add a warn log if it differs from `sourceOrderingField` set in top level config.

Reply:
and we need tests for this. 
for other payloads, if source ordering field is changed in subsequent commits, writes should fail.